hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java

```diff
@@ -105,6 +128,12 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         if (topic.hasSubmitKey()) {
             context.requireKeyOrThrow(topic.submitKeyOrThrow(), INVALID_SUBMIT_KEY);
         }
+        // add optional fee exempt keys in to the key verifieer
+        // later it will be used to validate if transaction was signed by
+        // any of these keys and based on that, custom fees will be charged or not
+        if (!topic.feeExemptKeyList().isEmpty()) {
+            context.optionalKeys(new HashSet<>(topic.feeExemptKeyList()));
```

Comment:
Was discussing with @tinker-michaelj  regarding this. This seems to be a lot of keys to verify, since we don't have limit on number of simple keys in the 10 keys allowed. Either we should add a limit on number of simple keys in `ConsensusCreateTopicHandler` or do this in a different way. @netopyr  thoughts?

Reply:
As discussed, we should do this check as @tinker-michaelj proposed.